impexp-core/src/main/java/org/citydb/core/operation/exporter/database/content/DBSplitter.java

```diff
@@ -238,16 +238,17 @@ private void queryCityObject(FeatureType cityObjectGroupType, Map<Long, DBSplitt
 			hits = getNumberMatched(query, connection);
 		}
 
+		// avoid long "where" clause in the root select statement
+		Table table = new Table(select);
+		select = new Select().addProjection(table.getColumn(MappingConstants.ID))
+				.addProjection(table.getColumn(MappingConstants.OBJECTCLASS_ID))
+				.addProjection(table.getColumn(MappingConstants.GMLID))
+				.addProjection(table.getColumn(MappingConstants.ENVELOPE));
```

Comment:
The `ENVELOPE` column may only be added if both `calculateExtent` and `query.isSetTiling()` are true (see previous code). Otherwise, the query will throw errors, for example, when tiling is not enabled.

Reply:
@clausnagel 

> @BWibo, have you already tried and "vacuumed" your database? Maybe this already helps?

I just did a `vacuum analyze` on the database and tested again with ImpExp `v5.2.0`. 
The export did not get stuck this time! I thought I had done this already, but apparently I forgot to run the vaccum.

If this resolves the issue with the original query, we should at least add a hint to vaccum the database, after importing large datasets.

Reply:
oh, puh, the query has problem with Oracle. Tested with both 5.1 and 5.2 and Oracle 12 and 19

```
[11:31:05 ERROR] Failed to query the database.
[11:31:05 ERROR] Caused by: java.sql.SQLException: ORA-22901: VARRAY oder LOB-Attribute eines Objekttyps können nicht verglichen werden
[11:31:05 ERROR] Caused by: oracle.jdbc.OracleDatabaseException: ORA-22901: VARRAY oder LOB-Attribute eines Objekttyps können nicht verglichen werden
[11:31:05 WARN] Database export aborted.
```

Reply:
Should we open a separate issue to deal with this error?

Reply:
yes, see #262